### PR TITLE
updates release workflow and gradle plugin

### PR DESCRIPTION
**Pull request by JBAhire**
## Description
addresses https://github.com/hypertrace/hypertrace/issues/19 & https://github.com/hypertrace/hypertrace/issues/22 

Please read discussion here: https://github.com/hypertrace/attribute-service/pull/30

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -83,6 +93,7 @@ jobs:
       - run:
           name: Helm Charts Lint and Template Render
           command: |
+            helm dependency update ./helm/
```
Comment (aaron-steinfeld) [inline .circleci/config.yml:96]: nit: this chart has no dependencies (and below)

**Review by aaron-steinfeld — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (JBAhire) [reply .circleci/config.yml]: removed

**Review by aaron-steinfeld — APPROVED** [review]

**Review by tim-mwangi — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -57,23 +60,30 @@ jobs:
...
-          args: :tag -Prelease
-      - add_ssh_keys:
-          fingerprints:
-            - '9e:5e:1b:28:cf:14:d6:7d:cd:48:64:72:d0:6f:01:c8'
-      - run: git push origin $(./gradlew -q :printVersion)
+          args: dockerPushImages
+  release-publish:
+    executor: gradle_docker
+    steps:
+      - setup_build_environment
```
Comment (kotharironak) [inline .circleci/config.yml:84]: Now, I see two different things - one with docker creds, and one without docker creds (https://github.com/hypertrace/hypertrace-ingester/pull/51). 
Not a blocker for this PR, but we need to see how we want to proceed on forked branch PR.

**Review by kotharironak — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (JBAhire) [reply .circleci/config.yml:84]: Let's take care of this in separate PR. I somehow missed this.

Comment (aaron-steinfeld) [reply .circleci/config.yml:84]: Commented on that PR - please don't propagate removing credentials until we come to an agreed upon approach.

**Pull request merged**